fixtures/coverall/tests/bindings/test_coverall.swift

```diff
@@ -0,0 +1,103 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import coverall
+
+// TODO: use an actual test runner.
+
+
+// Floats should be "close enough" for testing purposes.
+fileprivate extension Double {
+    func almostEquals(_ other: Double) -> Bool {
+        return abs(self - other) < 0.000001
+    }
+}
+
+fileprivate extension Float {
+    func almostEquals(_ other: Float) -> Bool {
+        return abs(self - other) < 0.000001
+    }
+}
+
+// Test some_dict().
+do {
+    let d = createSomeDict()
+    assert(d.text == "text")
+    assert(d.maybeText == "maybe_text")
+    assert(d.aBool)
+    assert(d.maybeABool == false);
+    assert(d.unsigned8 == 1)
+    assert(d.maybeUnsigned8 == 2)
+    assert(d.unsigned64 == 18446744073709551615)
+    assert(d.maybeUnsigned64 == 0)
+    assert(d.signed8 == 8)
+    assert(d.maybeSigned8 == 0)
+    assert(d.signed64 == 9223372036854775807)
+    assert(d.maybeSigned64 == 0)
+    assert(d.float32.almostEquals(1.2345))
+    assert(d.maybeFloat32!.almostEquals(22.0/7.0))
+    assert(d.float64.almostEquals(0.0))
+    assert(d.maybeFloat64!.almostEquals(1.0))
+    assert(d.coveralls!.getName() == "some_dict")
+}
+
+// Test arcs.
+do {
+    let coveralls = Coveralls(name: "test_arcs")
+    assert(getNumAlive() == 1)
+    // One ref held by the foreign-language code, one created for this method call.
+    assert(coveralls.strongCount() == 2)
+    assert(coveralls.getOther() == nil)
+    coveralls.takeOther(other: coveralls)
+    // Should now be a new strong ref, held by the object's reference to itself.
+    assert(coveralls.strongCount() == 3)
+    // But the same number of instances.
+    assert(getNumAlive() == 1)
+    // It's the same Rust object.
+    assert(coveralls.getOther()!.getName() == "test_arcs")
+    // TODO: ideally, it would return the same Swift-level object.
+    // assert(coveralls.getOther()!! === coveralls);
+    do {
+        try coveralls.takeOtherFallible()
+        fatalError("Should have thrown")
+    } catch CoverallError.TooManyHoles {
+        // It's okay!
+    }
+    // TODO: kinda hard to test this, as it triggers a fatal error.
+    // coveralls!.takeOtherPanic(message: "expected panic: with an arc!")
+    coveralls.takeOther(other: nil);
+    assert(coveralls.strongCount() == 2);
+}
+// Swift GC is deterministic, `coveralls` is freed when it goes out of scope.
+assert(getNumAlive() == 0);
```

Comment:
By contrast, the Swift GC is based on reference-counting so this is all totally deterministic \o/